### ISSUE TITLE
learning rate adjustment 

### ISSUE THREAD

**Issue by 1073521013** [qqingzheng/AI-Self-Training-DPO-SDXL#1]
Hello, thank you for your open source. How did you adjust the learning rate? The paper suggests 1e-8 while you used 1e-5, which is quite a big difference. Could you share your adjustment strategy(including some other hyperparameters) ?

**Comment by qqingzheng** [comment]
I also encountered difficulties here before. In the final training, I used a learning rate of 1e-06 and slightly modified the loss function (which was effective in practice but not proven). You can see the relevant code in the latest `train.py`.

**Comment by 1073521013** [comment]
Yes, I also noticed the difference in losses. Recently, I have been trying to adjust parameters, especially the learning rate, which feels difficult to adjust. Thank you for your reply



---Original---
From: ***@***.***&gt;
Date: Wed, Dec 20, 2023 20:08 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [qqingzheng/AI-Self-Training-DPO-SDXL] learning rate adjustment (Issue #1)




 
I also encountered difficulties here before. In the final training, I used a learning rate of 1e-06 and slightly modified the loss function (which was effective in practice but not proven).

**Comment by qqingzheng** [comment]
The loss chart I trained is here, hoping it can serve as a reference for you.
![W B Chart 2023_12_20 20_19_00](https://github.com/qqingzheng/AI-Self-Training-DPO-SDXL/assets/88202804/adc5cb69-5e9d-4a54-9fb9-7d93e86699f4)

**Comment by 1073521013** [comment]
Thank you very much, I will take a closer look tomorrow!




---Original---
From: ***@***.***&gt;
Date: Wed, Dec 20, 2023 20:20 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [qqingzheng/AI-Self-Training-DPO-SDXL] learning rate adjustment (Issue #1)




 
The loss chart I trained is here, hoping it can serve as a reference for you.
 W.B.Chart.2023_12_20.20_19_00.png (view on web)

**Comment by ToddBear** [comment]
> I also encountered difficulties here before. In the final training, I used a learning rate of 1e-06 and slightly modified the loss function (which was effective in practice but not proven). You can see the relevant code in the latest `train.py`.

Can you please provide us with the latest "train.sh" file? Because the learning rate in the current "train.sh" file is set as 1e-5, which is inconsistent with your latest reply.

Additionally, can you please provide more training details? such as the number of GPU for training, and the number of paired samples for training.

Thanks for your kindness!

**Comment by qqingzheng** [comment]
**Training Parameters**
The latest `train.sh` has only been modified with the learning rate compared to the `train.sh` in the repo. I will update the repository after organizing the code.
**Training Details**
8 80G A100 GPUs
**Dataset Details**
Num examples: 37180
Num Epochs: 3

Hoping these helps.

**Comment by ToddBear** [comment]
Thanks for your sharing!

Do you mind further sharing the datasets you used for training or the source of data used in "build_datasets.py"?

**Comment by qqingzheng** [comment]
> Thanks for your sharing!
> 
> Do you mind further sharing the datasets you used for training or the source of data used in "build_datasets.py"?

I will share the complete process of generating datasets in the near future. I am still organizing the code and doing some testing.
